experimental/visual-dataflow/src/CSVParser.h

```diff
@@ -0,0 +1,25 @@
+//===- CSVParser.h - Parses a csv transition line ------------*- C++ -*-===//
+//
+// The extracted data provides information about the state of an edge of the
+// graph for a given cycle .
+//===----------------------------------------------------------------------===//
+
+#ifndef VISUAL_DATAFLOW_CSVPARSER_H
+#define VISUAL_DATAFLOW_CSVPARSER_H
+#include "Graph.h"
+#include "mlir/Support/LLVM.h"
+#include "mlir/Support/LogicalResult.h"
+#include <iostream>
+
+namespace dynamatic {
+namespace experimental {
+namespace visual_dataflow {
+
+/// This function reads and interprets a edge transition line form a csv file.
+void processCSVLine(const std::string &line, Graph *graph, size_t lineIndex);
```

Comment:
Since `graph` should never be `nullptr` here it's better to pass it by reference (always better to pass by reference than by pointer if `nullptr` is not an acceptable value)

```cpp
void processCSVLine(const std::string &line, size_t lineIndex, Graph &graph);
```

Then inside the function replace `graph->` by `graph.`. Also move the `graph` to the last argument: by convention arguments passed by reference and internally modified are passed last. Think of it as: first I pass my actual inputs, then my inputs that are somehow "return values" too, since I will read their updated state from the function's caller.